src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -116,6 +114,11 @@ def load_arguments(self, _):
         c.argument('platform_fault_domain_count', type=int, help='Fault Domain count.')
         c.argument('validate', help='Generate and validate the ARM template without creating any resources.', action='store_true')
         c.argument('unmanaged', action='store_true', min_api='2016-04-30-preview', help='contained VMs should use unmanaged disks')
+
+    with self.argument_context('vm availability-set update') as c:
+        if self.supported_api_version(max_api='2016-04-30-preview', operation_group='virtual_machines'):
+            c.argument('name', name_arg_type, id_part='name', completer=get_resource_name_completion_list('Microsoft.Compute/availabilitySets'), help='Name of the availability set')
+            c.argument('availability_set_name', options_list=['--availability-set-name'])
```

Comment:
Don't know the knack syntax too much, but should we have an `else` here for `availability_set_name`?

Reply:
That's covered elsewhere. I could have called it anything since it doesn't appear (but ignore didn't work...annoying).  This is a very edge case scenario. So far I've never had to put an if statement in the Knack-based modules like this.